uhubctl.c

```diff
@@ -1051,9 +1058,12 @@ int main(int argc, char *argv[])
                 for (port=1; port <= hubs[i].nports; port++) {
                     if ((1 << (port-1)) & ports) {
                         int port_status = get_port_status(devh, port);
+                        printf("Port Status: %i\n", port_status);
+                        if (opt_action == POWER_TOGGLE)
+                            request = (port_status == 0) ? LIBUSB_REQUEST_SET_FEATURE : LIBUSB_REQUEST_CLEAR_FEATURE;
                         int power_mask = hubs[i].super_speed ? USB_SS_PORT_STAT_POWER
                                                              : USB_PORT_STAT_POWER;
-                        if (k == 0 && !(port_status & power_mask))
+                        if (k == 0 && !(port_status & power_mask) && (opt_action != POWER_TOGGLE))
```

Comment:
This condition does not seem right. I assume you meant this?

    if (k == 0 && (!(port_status & power_mask) || opt_action != POWER_TOGGLE))

Reply:
I don't think that this one is wrong, as it should only "continue", if power_toggle is not requested AND the other conditions are met.